core/container.go

```diff
@@ -560,6 +561,8 @@ func (container *Container) WithMountedFile(ctx context.Context, bk *buildkit.Cl
 	return container.withMounted(ctx, bk, target, file.LLB, file.File, file.Services, owner)
 }
 
+var SeenCacheKeys sync.Map
```

Comment:
nit: I believe by convention in the codebase, this should be achieved by passing *something* in core/schema, and having that something passed again to `containerSchema` (just like the `leaseManager`, `buildCache`, etc) rather than using a global variable.

/cc @vito who added the most in there recently

Reply:
I looked into this and the thing is that I'd have to pass this "_something_" across multiple layers so it gets all the way to the `schema` package, thing is that the `CacheManager` defined here (https://github.com/marcosnils/dagger/blob/a1e35f20a546f00c4e4fcc13581b1ab833a1640d/cmd/engine/main.go#L760) which has the mount synchronization logic a methodso gets passed as a `solver.CacheManager` here (https://github.com/marcosnils/dagger/blob/a1e35f20a546f00c4e4fcc13581b1ab833a1640d/cmd/engine/main.go#L802) to the buildkit controller and there's not to many things we could do from there. Additionally, the cache mount synchronization process happens in the `main` package here (https://github.com/marcosnils/dagger/blob/a1e35f20a546f00c4e4fcc13581b1ab833a1640d/cmd/engine/main.go#L802) which doesn't have way to access the schema or core structs to fetch the seen cache mount keys :shrug:

Reply:
I think this is OK as a special-case since this is really an engine-wide concern. We'll see if anything changes once we figure out cache namespacing/etc.